Source/Core/SphericalExtentsGeometryInstanceAttribute.js

```diff
@@ -0,0 +1,143 @@
+        './ComponentDatatype',
+        './defineProperties',
+        './Ellipsoid'
+    ], function(
+        Cartesian2,
+        Cartesian3,
+        Cartographic,
+        Check,
+        CesiumMath,
+        ComponentDatatype,
+        defineProperties,
+        Ellipsoid) {
+    'use strict';
+
+    function approximateSphericalLatitude(spherePoint) {
+        // Project into plane with vertical for latitude
+        var magXY = Math.sqrt(spherePoint.x * spherePoint.x + spherePoint.y * spherePoint.y);
+        return CesiumMath.fastApproximateAtan2(magXY, spherePoint.z);
+    }
+
+    function approximateSphericalLongitude(spherePoint) {
+        return CesiumMath.fastApproximateAtan2(spherePoint.x, spherePoint.y);
+    }
+
+    var cartographicScratch = new Cartographic();
+    var cartesian3Scratch = new Cartesian3();
+    function latLongToSpherical(latitude, longitude, result) {
+        var carto = cartographicScratch;
+        carto.latitude = latitude;
+        carto.longitude = longitude;
+        carto.height = 0.0;
+
+        var cartesian = Cartographic.toCartesian(carto, Ellipsoid.WGS84, cartesian3Scratch);
+        var sphereLatitude = approximateSphericalLatitude(cartesian);
+        var sphereLongitude = approximateSphericalLongitude(cartesian);
+        result.x = sphereLatitude;
+        result.y = sphereLongitude;
+
+        return result;
+    }
+
+    var sphericalScratch = new Cartesian2();
+
+    /**
+     * Spherical extents needed when computing ground primitive texture coordinates per-instance.
+     * Used for "large distances."
+     * Computation is matched to in-shader approximations.
+     *
+     * Consists of western and southern spherical coordinates and inverse ranges.
+     *
+     * @alias SphericalExtentsGeometryInstanceAttribute
+     * @constructor
+     *
+     * @param {Rectangle} rectangle Conservative bounding rectangle around the instance.
+     *
+     * @see GeometryInstance
+     * @see GeometryInstanceAttribute
+     * @see createShadowVolumeAppearanceShader
+     * @private
+     */
+    function SphericalExtentsGeometryInstanceAttribute(rectangle) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.typeOf.object('rectangle', rectangle);
+        //>>includeEnd('debug');
+
+        // rectangle cartographic coords !== spherical because it's on an ellipsoid
+        var southWestExtents = latLongToSpherical(rectangle.south, rectangle.west, sphericalScratch);
+
+        // Slightly pad extents to avoid floating point error when fragment culling at edges.
+        var south = southWestExtents.x - CesiumMath.EPSILON5;
+        var west = southWestExtents.y - CesiumMath.EPSILON5;
+
+        var northEastExtents = latLongToSpherical(rectangle.north, rectangle.east, sphericalScratch);
+        var north = northEastExtents.x + CesiumMath.EPSILON5;
+        var east = northEastExtents.y + CesiumMath.EPSILON5;
+
+        var longitudeRangeInverse = 1.0 / (east - west);
```

Comment:
Do these need to worry about divide by zero?

Reply:
In practice they shouldn't, since these are used for "large areas."